extensions/wikia/Piggyback/Piggyback_body.php

```diff
@@ -12,12 +11,23 @@ function __construct() {
 		global $wgRequest;
 		parent::__construct( 'Piggyback', 'piggyback' );
 		$this->mAction = $wgRequest->getVal( 'action' );
-
 		$this->logger = Wikia\Logger\WikiaLogger::instance();
 	}
 
 	function execute( $par ) {
-		global $wgRequest, $wgOut, $wgUser;
+		global $wgRequest, $wgOut, $wgUser, $wgEnableMercuryPiggyback;
+
+		$this->logger->info( 'IRIS-4219 Piggyback has been rendered' );
+
+		if ( !empty( $par ) ) {
+			$wgRequest->setVal( 'target', $par );
+		}
+
+		if ( $wgEnableMercuryPiggyback ) {
+			$this->redirectToMercuryPiggyback( $wgOut, $wgRequest->getVal( 'target' ) );
```

Comment:
You can use the local `RequestContext` here instead of globals, i.e. `$this->getRequest()` and `$this->getOutput()`